apps/server/src/infra/sync/service/sync.service.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable, Optional } from '@nestjs/common';
+import { TspSyncStrategy } from '../tsp/tsp-sync.strategy';
+import { SyncStrategy } from '../strategy/sync-strategy';
+import { SyncStrategyTarget } from '../sync-strategy.types';
+
+@Injectable()
+export class SyncService {
+	private strategies: Map<SyncStrategyTarget, SyncStrategy> = new Map<SyncStrategyTarget, SyncStrategy>();
+
+	constructor(@Optional() private readonly tspSyncStrategy?: TspSyncStrategy) {
+		this.registerStrategy(tspSyncStrategy);
+	}
+
+	protected registerStrategy(strategy?: SyncStrategy) {
+		if (strategy) {
+			this.strategies.set(strategy.getType(), strategy);
+		}
+	}
+
+	public async startSync(target: string): Promise<void> {
+		const targetStrategy = target as SyncStrategyTarget;
+		if (!this.strategies.has(targetStrategy)) {
+			throw new Error('please provide a valid target strategy name to start its synchronization process');
```

Comment:
Maybe make this an Loggable to not have a stacktrace in the output

Reply:
done ✅